### ISSUE TITLE
Backup Incomplete state in GCS bucket with MD5 checksum error

### ISSUE THREAD

**Issue by jralmaraz** [thelastpickle/cassandra-medusa#564]
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=20774320)

Hi,

I would like to find more on the root cause of an existing backup issue and the best course of action to get the states back to normal (maybe cleanup the bucket and run a full fresh backup ?).

[medusa-md5-error.txt](https://github.com/thelastpickle/cassandra-medusa/files/10767968/medusa-md5-error.txt)

Basically there 3 nodes in the cluster and I notice in the verify output a mention of a hostname of a fourth instance that is not part of the cluster - so I wonder if there is maybe some miss that may be causing this inconsistency.

Any pointers are welcome.

Thanks a lot!

Jose

**Comment by adejanovski** [comment]
Hi @jralmaraz, which version of Medusa are you using?
The problem you're seeing is not related to that fourth node, it's the comparison for each node of the backup manifest with what's actually in storage that fails.

```
[2023-02-17 12:10:40,530] ERROR: Expected {'path': 'cluster1/instance3/20230217/data/bla/bla-b266c58032b111e88126b7343b599b7c', 'MD5': '', 'size': 393} but got <Object: name=cluster1/instance3/20230217/data/bla/bla-b266c58032b111e88126b7343b599b7c, size=48, hash=-CJTnn/S7m/0CEAE=, provider=Google Cloud Storage ...>
```

From that line, it looks like the `bla-b266c58032b111e88126b7343b599b7c` object is seen as a file instead of a folder (which it should be).
This reminds me of an issue that was fixed already: https://github.com/thelastpickle/cassandra-medusa/pull/205
Do you have access to one of these files and could tell me what their content looks like? It seems like there's possibly a new type of file that we should exclude from the backup to prevent this from happening again.
